src/ordered_writer.c

```diff
@@ -0,0 +1,62 @@
+#include <assert.h>
+#include <roaring/roaring.h>
+
+roaring_bitmap_writer_t *roaring_bitmap_writer_create(roaring_bitmap_t *target) {
+    roaring_bitmap_writer_t *result = (roaring_bitmap_writer_t *) malloc(sizeof(roaring_bitmap_writer_t));
+    result->bitmap = (uint64_t *) malloc(sizeof(uint64_t) * BITSET_CONTAINER_SIZE_IN_WORDS);
+    memset(result->bitmap, 0, sizeof(uint64_t) * BITSET_CONTAINER_SIZE_IN_WORDS);
+    result->target = target;
+    result->current_key = 0;
+    result->cardinality = 0;
+    result->dirty = false;
+    return result;
+}
+
+void roaring_bitmap_writer_free(roaring_bitmap_writer_t *writer) {
+    free(writer->bitmap);
+    free(writer);
+}
+
+void roaring_bitmap_writer_flush(roaring_bitmap_writer_t *writer) {
+    if (!(writer->dirty)) {
+        return;
+    }
+
+    uint64_t *bitmap = writer->bitmap;
+    roaring_bitmap_t *target = writer->target;
+
+    bitset_container_t *newac = bitset_container_create();
+
+    newac->cardinality = -1;
+    memcpy(newac->array, bitmap, sizeof(uint64_t) * BITSET_CONTAINER_SIZE_IN_WORDS);
+
+    uint8_t new_typecode = BITSET_CONTAINER_TYPE_CODE;
+    void *newcontainer = container_repair_after_lazy(newac, &new_typecode);
+
+    ra_append(&target->high_low_container, writer->current_key, newcontainer, new_typecode);
```

Comment:
This will work only if roaring bitmap was empty.

Instead we can do the following:
1. Redefine `roaring_bitmap_writer_s` as following:

       typedef struct roaring_bitmap_writer_s {
         roaring_bitmap_t *target;
         bitmap_container_t* tmp_container;
         uint16_t current_key;
       };
2. Every time `roaring_bitmap_writer_add()` is called, add given value directly into `tmp_container`
3. When time comes to flush, get container from `writer->target` with the key `writer->current_key`
    * If such container exists, then call `container_ior` with `tmp_container`
    * If it doesn't exist, then insert `tmp_container` into roaring_bitmap.

Reply:
The point is to avoid inserting right in the containers for every value to avoid reallocation, but it's possible to achieve this If I create bitset_container_t and use this logic, right?